source/common/stats/allocator_impl.cc

```diff
@@ -194,12 +214,23 @@ class GaugeImpl : public StatsSharedImpl<Gauge> {
   }
 };
 
+template <class StatType> static void warnOnManyUses(StatType& stat) {
+  // This is called when we are about to increment the ref-count, so use that
+  // incremented number in the arithmetic. Issue a warning message only when
+  // the use-count gets very high, and then only once in 10k uses.
+  uint64_t use_count = stat.use_count() + 1;
+  if ((use_count % 10000) == 0) {
+    ENVOY_LOG_MISC(warn, "{}: very high use-count: {}", stat.name(), use_count);
+  }
+}
```

Comment:
FWIW I think I would probably just kill this, given that I don't think it's going to be obvious to monitor this operationally and also if this does fire, it may very well log spam. So if we do want this IMO we need to go ahead and add some type of ENVOY_LOG_ONCE or ENVOY_LOG_RATE_LIMIT macro which we have discussed many times at this point. WDYT?

Reply:
i'll kill it from here and put it in the test. I don't want to get stuck in the mud of the logging macros before fixing this bug.